### PR TITLE
Remove pallet::getter usage from treasury

### PR DESCRIPTION
**Pull request by Polkaverse**
ISSUE
Link to the issue: https://github.com/paritytech/polkadot-sdk/issues/3326

Deliverables

[Deprecation] remove pallet::getter usage from pallet-treasury

### CODE REVIEW

File: substrate/frame/treasury/src/lib.rs [hunk 1]
```diff
@@ -276,12 +276,10 @@ pub mod pallet {
 
 	/// Number of proposals that have been made.
 	#[pallet::storage]
-	#[pallet::getter(fn proposal_count)]
 	pub(crate) type ProposalCount<T, I = ()> = StorageValue<_, ProposalIndex, ValueQuery>;
```
Comment (ggwpez) [inline substrate/frame/treasury/src/lib.rs:279]:
```suggestion
	pub type ProposalCount<T, I = ()> = StorageValue<_, ProposalIndex, ValueQuery>;
```

**Review by ggwpez — APPROVED** [review]

**Comment by Polkaverse** [comment]
@ggwpez @bkchr

File: prdoc/pr_4962.prdoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Removed `pallet::getter` usage from the pallet-treasury
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR removed `pallet::getter`s from `pallet-treasury`s storage items.
+      When accessed inside the pallet, use the syntax `StorageItem::<T, I>::get()`.
```
Comment (seadanda) [inline prdoc/pr_4962.prdoc:10]:
```suggestion
      Instead use the syntax `StorageItem::<T, I>::get()`.
```

This works outside the pallet too.

File: prdoc/pr_4962.prdoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+crates:
+  - name: pallet-treasury
+    bump: minor
```
Comment (seadanda) [inline prdoc/pr_4962.prdoc:14]: This is surely a major change
```suggestion
    bump: major
```

**Review by seadanda — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

File: prdoc/pr_4962.prdoc (hunk 3 above)
Comment (bkchr) [reply prdoc/pr_4962.prdoc]: Is it not

Comment (bkchr) [reply prdoc/pr_4962.prdoc]: The same functions still exist.

File: prdoc/pr_4962.prdoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR removed `pallet::getter`s from `pallet-treasury`s storage items.
+      Instead use the syntax `StorageItem::<T, I>::get()`.
+
+crates:
+  - name: pallet-treasury
+    bump: major
```
Comment (bkchr) [inline prdoc/pr_4962.prdoc:14]:
```suggestion
    bump: minor
```

File: prdoc/pr_4962.prdoc (hunk 3 above)
Comment (seadanda) [reply prdoc/pr_4962.prdoc:14]: 🤦‍♂️

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/sponsorship-for-sub0-thailand-2024/10093/37

**Pull request merged**